src/_pytest/terminal.py

```diff
@@ -483,43 +484,51 @@ def pytest_runtest_logreport(self, report: TestReport) -> None:
 
     @property
     def _is_last_item(self):
-        return len(self._progress_nodeids_reported) == self._session.testscollected
+        return self._progress_items_reported == self._session.testscollected
 
-    def pytest_runtest_logfinish(self, nodeid):
-        assert self._session
-        if self.verbosity <= 0 and self._show_progress_info:
-            if self._show_progress_info == "count":
-                num_tests = self._session.testscollected
-                progress_length = len(" [{}/{}]".format(str(num_tests), str(num_tests)))
-            else:
-                progress_length = len(" [100%]")
+    def pytest_runtest_teardown(self) -> None:
```

Comment:
I'm curious, why using `pytest_runtest_teardown` instead of `pytest_runtest_logfinish`? On pytest-xdist I think `pytest_runtest_teardown` is called only on worker nodes (because they actually run the tests), not on `master` (which only does reporting). The `pytest_runtest_log*` run on both `master` and `workers`, by design.

Reply:
Hmmm strange, I commented out this function from your branch, and I still get the exact same output both with xdist/without, and `-v` and normal modes.

---

With the patch as is:

```
λ pytest testing\test_nodes.py
======================== test session starts ========================
collected 11 items

testing\test_nodes.py ...........                              [100%]

======================== 11 passed in 0.10s =========================
```

Commenting out `pytest_runtest_teardown`:

```
 λ pytest testing\test_nodes.py
======================== test session starts ========================
collected 11 items

testing\test_nodes.py ...........                              [100%]

======================== 11 passed in 0.10s =========================
```

---

With the patch as is:

```
λ pytest testing\test_nodes.py -n2
======================== test session starts ========================
gw0 [11] / gw1 [11]
...........                                                    [100%]
======================== 11 passed in 0.93s =========================
```

Commenting out `pytest_runtest_teardown`:

```
λ pytest testing\test_nodes.py -n2
======================== test session starts ========================
gw0 [11] / gw1 [11]
...........                                                    [100%]
======================== 11 passed in 0.92s =========================
```

---

So I guess `pytest_runtest_teardown` is not being used?

I got curious, and when in `master` and `pytest_runtest_logfinish` commented out, the `[%]` progress indicator is not printed.

Reply:
> The pytest_runtest_log* run on both master and workers, by design.

That might result in too much being reported then (if both master and nodes report the same item(s)).
Changed it to use pytest_runtest_logfinish again, let's see if it passes.